### ISSUE TITLE
Autotile all windows in workspace using a resizing preset

### ISSUE THREAD

**Issue by hermidalc** [gTile/gTile#87]
Not sure if feature is already available or how I would be able to do it using existing autotile and configurable features.  I would like to be able to autotile all windows in the workspace using one of the resizing presets.  Is that possible?

**Comment by scherepanov** [comment]
Yes, you can force screen autotile with keyboard shortcut. It is described in README.md. Please check and let me know if it is not described properly.
In short, you activate gTile - by keyboard shortcut Super-KP-Enter. It activate additional shortcuts Super-KP-1 - 9.If you use them, gTile will do autotile for all windows on a screen, in two or more columns.

**Comment by hermidalc** [comment]
But how can I autotile all the windows in the workspace to my custom "Resize Preset 1" for example? It doesn't work if you use Super-KP_1-9 those are for other tiling operations than the custom resize presets.

**Comment by scherepanov** [comment]
It is not clear to me exactly what you want.
You want to tile all windows to some grid, one window per tile?
"Resize Preset 1" can be set to rather arbitrary config. For example, center 4 tiles (in 2x2 square) on a grid of 8x6.
In this case, where would other window will be going during autotile?

**Comment by hermidalc** [comment]
> You want to tile all windows to some grid, one window per tile?

Yes

> "Resize Preset 1" can be set to rather arbitrary config. For example, center 4 tiles (in 2x2 square) on a grid of 8x6.
> In this case, where would other window will be going during autotile?

For example here's my Resize Preset 1: 7x9 0:0 2:4,3:0 6:4,0:5 2:8,3:5 6:8

I want to be in a workspace with 4 windows and then autotile them to that preset one per tile in one keystroke.  How can I do that?

**Comment by KristerV** [comment]
yeah this is super confusing. As I understand currently the "resize preset" is when you want to resize a single window to a preset. Autotile is when you want to resize all the windows, yet it just doesn't work.. gnome title bar flashes for a sec (probably due to autohide), but nothing actually changes.

**Comment by scherepanov** [comment]
@KristerV  it is an error in docs. You need to use super + KB_enter plus super + 3
On second one number is not on Keypad.
Thanks for raising attention to this.
@hermidalc you are requesting new feature, and it need to be discussed how usable for large user population it would be. So far I am sceptical that it will be a lot of usage of this new feature.

**Comment by KristerV** [comment]
yes, this works (i thought i tried it actually). this plugin is awesome!

**Comment by hermidalc** [comment]
> @hermidalc you are requesting new feature, and it need to be discussed how usable for large user population it would be. So far I am sceptical that it will be a lot of usage of this new feature.

Here's my general work use case (which I thought would be common):

I put soft dev related windows in one workspace, terminals in one, and internet related in another.  In each workspace I like to tile up the same way where particular apps/windows get a certain size and location in the workspace.  With X I would have a bash script to launch, size and position most automatically.  Though since Wayland I believe you cannot position windows automatically anymore so I've been looking for some extension to help me so I don't have to manually do it.  Do other people have a similar use case?

I'm a new gTile user and it's the closest thing I've seen and it would be great if it could be configured in such a way with the resize presets.

**Comment by scherepanov** [comment]
I think general use case can be as follow.
You put focus on some window, and autotile rest. Focus window is not being moved. Rest of windows on a screen is autotiled around focus window.
And, autotile works by arranging windows in several columns.
That can be as close as I think about generic use case.
Another optoin would be to force autotile into current grid, one window per tile. There is a "current grid" that is selected when gTile window show up. You can cycle through available grids by pressing space.
Second option is not linked to any presets.

**Comment by jabberwock7** [comment]
I agree that there should be a button or easy way to autotile an entire screen using the grid system selected.

**Comment by scherepanov** [comment]
@jabberwock7 what do you think is more useful out of two autotile options:

You put focus on some window, and autotile rest. Focus window is not being moved. Rest of windows on a screen is autotiled around focus window.
And, autotile works by arranging windows in several columns.
That can be as close as I think about generic use case.
Another optoin would be to force autotile into current grid, one window per tile. There is a "current grid" that is selected when gTile window show up. You can cycle through available grids by pressing space.
Second option is not linked to any presets.

**Comment by TyrelCB** [comment]
Only been using gTile for a couple hours now, great extension. I agree with @hermidalc that it would be really cool to use the "Autotile Main" function or something similar with a custom "Preset Resize". Instead of having to select each window and having to cycle through the presets, for example `8x6 0:0 4:5, 8x10 5:0 7:3, 8x22 5:9 7:21`
